src/util/middlewares.js

```diff
@@ -20,10 +20,12 @@ async function errorMiddleware(err, req, res, _next) {
  * @param {import('express').NextFunction} next
  */
 const validateUserMiddleware = async (req, res, next) => {
-  let user = await validateUser(req, res);
-  if (!user) {
+  let session = await validateUser(req, res);
+  if (!session) {
     return;
   }
+
+  req.userSession = session;
```

Comment:
I tried to implement a simple solution. there is https://github.com/expressjs/session express-session. It looks overkill. Never used it but if you understand that this is the direction to go I can learn how to use and go from there.

Reply:
So, I investigated express-session and it basically is an overhead (for now) for what we are trying to achieve.
The way we add user data to session is basically the same as we are doing:
![image](https://github.com/user-attachments/assets/ea0de898-4fb9-4bbb-86be-998f6bbf449e)
this is the doc from express-session on how to add custom data into session.

It makes sense when using typescript, but since server is just javascript. I suggest to keep the current implementation and when we move the server from js to ts we implement a more robust typechecking and session handler as express-session. WDYT @MatissJanis ?

Reply:
changed to res.locals